modules/imgproc/src/sumpixels.avx512_skx.cpp

```diff
@@ -0,0 +1,260 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019, Intel Corporation, all rights reserved.
+#include "precomp.hpp"
+#include "sumpixels.hpp"
+
+namespace cv {
+namespace opt_AVX512_SKX {
+
+//
+// NOTE: Look at the bottom of the file for the entry-point function for external callers
+//
+
+// At the moment only 3 channel support untilted is supported
+// More channel support coming soon.
+// TODO: Add support for sqsum and 1,2, and 4 channels
+class IntegralCalculator_3Channel {
```

Comment:
IMHO it would be better to place the class inside anonymous namespace to avoid its exposure outside of the module

Reply:
Ah - thanks for the clarification Vitaly.  This makes perfect sense: protect the implementation, expose only the entry method.

Done and pushed.